tests/ZonedDateTimeTest.php

```diff
@@ -482,4 +486,438 @@ public function providerCompareTo() : array
+
+        $this->assertSame(20, $zonedDateTime->getDay());
+    }
+
+    public function testGetDayOfWeek()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertDayOfWeekIs(4, $zonedDateTime->getDayOfWeek());
+    }
+
+    public function testGetDayOfYear()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame(20, $zonedDateTime->getDayOfYear());
+    }
+
+    public function testGetHour()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame(12, $zonedDateTime->getHour());
+    }
+
+    public function testGetMinute()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame(34, $zonedDateTime->getMinute());
+    }
+
+    public function testGetSecond()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame(56, $zonedDateTime->getSecond());
+    }
+
+    public function testWithDate()
+    {
+        $y = (int)date('Y');
+        $m = (int)date('m');
+        $d = (int)date('d');
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $newDate = LocalDate::of($y, $m, $d);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame(date('Y-m-d'), (string)$zonedDateTime->withDate($newDate)->getDate());
+    }
+
+    public function testWithTime()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $time = LocalTime::of(12, 34, 56, 987654321);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame('12:34:56.987654321', (string)$zonedDateTime->withTime($time)->getTime());
```

Comment:
Same as above, you should check the whole zoned date-time string because we need to ensure that the date is correct, as well.